src/toil/spark/spark_cluster.py

```
@@ -0,0 +1,328 @@
+        """
+        Start spark and hdfs master containers
+
+        fileStore: Unused
+        """
+        
+        self.IP = subprocess.check_output(["hostname", "-f",])[:-1]
+
+        _log.info("Started Spark master container.")
+        self.sparkContainerID = _docker_call(no_rm = True,
+                                             work_dir = os.getcwd(),
+                                             tool = "quay.io/ucsc_cgl/apache-spark-master:1.5.2",
+                                             docker_parameters = ["--net=host",
+                                                                  "-d",
+                                                                  "-v", "/mnt/ephemeral/:/ephemeral/:rw",
+                                                                  "-e", "SPARK_MASTER_IP="+self.IP,
+                                                                  "-e", "SPARK_LOCAL_DIRS=/ephemeral/spark/local",
+                                                                  "-e", "SPARK_WORKER_DIR=/ephemeral/spark/work"],
+                                             tool_parameters = [],
+                                             sudo = self.sudo,
+                                             check_output = True)[:-1]
+        _log.info("Started HDFS Datanode.")
+        self.hdfsContainerID = _docker_call(no_rm = True,
+                                            work_dir = os.getcwd(),
+                                            tool = "quay.io/ucsc_cgl/apache-hadoop-master:2.6.2",
+                                            docker_parameters = ["--net=host",
+                                                                 "-d"],
+                                            tool_parameters = [self.IP],
+                                            sudo = self.sudo,
+                                            check_output = True)[:-1]
+
+        # attach worker service jobs
+        self._add_worker_services()
```

Oops, this addition of the workers should happen in the constructor, after the call to the Job.Service parent class constructor. I need to make the docs and design clearer on this point. Minimimally it should throw an exception if you try to add a child service.in the start method.